### PR TITLE
initial challenges

### PR DESCRIPTION
**Pull request by alastairdglennie**
Couldn't find the medal icon so placeholder for now:

image

image

image

image

### CODE REVIEW

File: src/components/PageWrapper.tsx [hunk 1]
```diff
@@ -273,6 +276,9 @@ const PageWrapper: FunctionComponent<Props> = ({
           <Typography className={classes.grow} variant="h6" color="inherit">
             {label}
           </Typography>
+          {addAction !== undefined && (
+              <AddIcon onClick={addAction} />
```
Comment (neilfulwiler) [inline src/components/PageWrapper.tsx:280]: should we wrap this in an IconButton and attach the onClick handler to that?

File: src/types/Challenges.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import {StatsUser} from "./Stats";
+
+type Challenge = {
```
Comment (neilfulwiler) [inline src/types/Challenges.ts:3]: can inherit `NewChallengeRequest`?

File: src/routes/photo/routes/new/Route.tsx [hunk 3]
```diff
@@ -33,29 +35,12 @@ export default function NewPhotoRoute() {
...
-            // that we loaded is from the camera, but for now just check that
-            // the lastModified date is < 30s ago
-            const fileDate = file.lastModified;
-            const ageInMinutes = (new Date().getTime() - fileDate) / 1000 / 60;
-            const imgFromCamera = isNaN(ageInMinutes) || ageInMinutes < 0.5;
-            handlePhotoSelect(file, imgFromCamera);
-          }
-        }}
-      />
+      <DesktopPhotoFallback ref={desktopPhotoRef}
```
Comment (neilfulwiler) [inline src/routes/photo/routes/new/Route.tsx:42]: +1

File: src/providers/ChallengesProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import Challenge from "../types/Challenges";
+import exampleImage from "assets/images/example.jpeg";
+
+const FakeChallenge: Challenge = {
+    id: 123,
+    name: "This is the CHALLENGE",
+    description: "this is the description of the challenge omg I hate plastic",
```
Comment (neilfulwiler) [inline src/providers/ChallengesProvider.tsx:7]: I think we should just hardcode this for all of them

File: src/pages/challenges/CreateChallenge.tsx [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+                </div>
+
+                <div className={classes.dateWrapper}>
+                    <div className={classes.dateLabel}>End date</div>
+                    <input className={classes.dateInput}
+                           type="date"
+                           onChange={(e) => setEndDate(new Date(e.currentTarget.value))}/>
+                    {(startDate !== null && endDate !== null && endDate < startDate) &&
+                    <div className={classes.inputWarning}>
+                        Start date cannot be in the past
```
Comment (neilfulwiler) [inline src/pages/challenges/CreateChallenge.tsx:269]: nit: this input warning is wrong

File: src/pages/challenges/CreateChallenge.tsx [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+
+    const today = new Date();
+
+    const history = useHistory();
+    const handleBack = {handleBack: () => history.goBack(), confirm: true};
+
+
+    const [name, setName] = useState("");
+    const [description, setDescription] = useState("");
+    const [targetPieces, setTargetPieces] = useState("0");
```
Comment (neilfulwiler) [inline src/pages/challenges/CreateChallenge.tsx:179]: why is this a string?

File: src/pages/challenges/ChallengesHome.tsx [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        return challenges;
+    }
+
+    return challenges.filter(challenge => challenge.name.includes(searchPrefix));
+}
+
+export default function ChallengesHome({handleClose, challenges}: Props) {
+    const history = useHistory();
+    const classes = useStyles();
+    const [searchPrefix, setSearchPrefix] = useState("")
```
Comment (neilfulwiler) [inline src/pages/challenges/ChallengesHome.tsx:70]: super nit: not actually used as a prefix

**Review by neilfulwiler — COMMENTED** [review]
amazing :) given the config, happy for you to merge this whenever its in a good state

File: src/pages/challenges/CreateChallenge.tsx [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+                <div className={classes.privateToggleWrapper}>
+                    <input type="radio"
+                           value={"Private challenge"}
+                           checked={isPrivate}
+                           onClick={() => setIsPrivate(!isPrivate)}/>
+                    Private challenge
+                </div>
+
+                <Button className={classes.submitButton}
+                        onClick={() => history.push(linkToAddChallengeCoverPhotoDialog())}
```
Comment (neilfulwiler) [inline src/pages/challenges/CreateChallenge.tsx:289]: this just adds the cover photo right? nothing actually "submits" the challenge?

File: src/components/PageWrapper.tsx (hunk 1 above)
Comment (alastairdglennie) [reply src/components/PageWrapper.tsx:280]: Happy to, the button/close buttons weren't wrapped so I did like this - should they be also?

File: src/pages/challenges/CreateChallenge.tsx (hunk 6 above)
Comment (alastairdglennie) [reply src/pages/challenges/CreateChallenge.tsx:179]: Was just for easy use of the commonly shared setter function, but I've changed to a number / refactored the setter

File: src/components/PageWrapper.tsx (hunk 1 above)
Comment (neilfulwiler) [reply src/components/PageWrapper.tsx:282]: ah fair enough, lets keep it then, not sure what the right way is but surely it can't be too different

**Review by neilfulwiler — APPROVED** [review]

**Comment by joe-mcguinness** [comment]
medal-solid
This is the medal @alastairdglennie ! let me know if it's an ok size

**Pull request merged**